client/src/App.jsx

```diff
@@ -1,13 +1,83 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import MapView from './mapView';
 import Interface from './Interface';
+import fetchParklets from '../utils/fetchParklets';
 
-const App = () => (
-  <div id="home">
-    <MapView />
-    <Interface />
-  </div>
-);
+const PARKLETS_ENDPOINT = 'https://data.sfgov.org/resource/6a7x-cttf.json';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      activeParklets: [],
+      hiddenParklets: [],
+      error: '',
+    };
+
+    this.handleFilters = this.handleFilters.bind(this);
+  }
+
+  componentDidMount() {
+    // IMPORTANT: Until db is implemented, fake data will be used to generate flags
+    // for a parklet that is open, has coffee, or wifi by RNG, this will change for
+    // every refresh of the page
+    fetchParklets(PARKLETS_ENDPOINT)
+      .then((data) => {
+        // *******FAKE DATA GENERATOR************
+        const random = () => Math.random() > 0.5;
+
+        data.forEach((d) => {
+          const node = d;
+          node.wifi = random();
+          node.food = random();
+          node.open = random();
+        });
+        // remove above function once graphQL db is configured
+
+        this.setState({ activeParklets: data });
+      })
+      .then(() => this.handleFilters())
+      .catch(err => this.setState({ error: err }));
+  }
+
+  handleFilters() {
+    const open = document.querySelector('.open');
```

Comment:
It isn't a direct manipulation since it's a state change that would result in calling render() again, but I see where you're coming from. This approach uses the existence of a class, that is if the FilterButtons are .btn-dark-outline or not to attempt to hide the parklet or send it to the else case, where we'll show it. The selectors are merely just to find out the event.

That said, I can always use states to see whether or not open, food, and wifi filters are active. This would seem cleaner. Thanks for the suggestion!

Reply:
It's a conflict because it's not separating concerns. In react, you can still access classNames via the event object if you really need them. It's better to use props though because it has much clearer data flow rather than reading the Dom from react.

Reply:
Also, while you're still using a state method to set the state, you're directly manulating the Dom by using `document.querySelector`